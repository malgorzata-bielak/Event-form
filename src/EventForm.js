import React from "react";
import DatePicker from "react-datepicker";
import "react-datepicker/dist/react-datepicker.css";

import Input from "./Input";
import SubmitButton from "./SubmitButton";

const EventForm = ({ onChange, date, onDateChange }) => (
  <form autoComplete="off">
    <Input labelName="First name" id="firstname" event={onChange} />

    <Input labelName="Last name" id="lastname" event={onChange} />

    <Input labelName="Email" id="email" event={onChange} />

    <DatePicker selected={date} onChange={onDateChange} />

    <SubmitButton />
  </form>
);

export default EventForm;
