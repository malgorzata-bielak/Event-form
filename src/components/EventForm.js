import React from "react";
import DatePicker from "react-datepicker";
import "react-datepicker/dist/react-datepicker.css";
import styled from "@emotion/styled";

import Input, { Label, textField } from "./Input";
import SubmitButton from "./SubmitButton";

const Form = styled.form`
  width: 100%;

  @media (min-width: 720px) {
    width: 432px;
  }
`;

const DatePickerStyle = styled(DatePicker)`
  ${textField}
  width: 120px;
`;

const EventForm = ({ onChange, onDateChange, onSubmit, date }) => (
  <Form autoComplete="off" onSubmit={onSubmit} id="user-data-form">
    <Input labelName="First name" id="firstname" event={onChange} autofocus />

    <Input labelName="Last name" id="lastname" event={onChange} />

    <Input labelName="Email" id="email" event={onChange} />

    <Label id="date-picker">
      Date
      <DatePickerStyle selected={date} onChange={onDateChange} />
    </Label>
    <SubmitButton />
  </Form>
);

export default EventForm;
