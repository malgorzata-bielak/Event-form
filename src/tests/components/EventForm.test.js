import React from "react";
import { shallow } from "enzyme";

import EventForm from "../../components/EventForm";

const propsData = {
  onChange: jest.fn(),
  onDateChange: jest.fn(),
  onSubmit: jest.fn(),
  date: 1558615381720
};

test("should render EventForm correctly", () => {
  const wrapper = shallow(<EventForm {...propsData} />);
  expect(wrapper).toMatchSnapshot();
});

test("should set new date on date change", () => {
  const wrapper = shallow(<EventForm {...propsData} />);
  const date = new Date().valueOf();

  wrapper.find("Styled(DatePicker)").simulate("change", date);
  expect(propsData.onDateChange).toHaveBeenCalledWith(date);
});

test("should call onSubmit prop for valid form submission", () => {
  const wrapper = shallow(<EventForm {...propsData} />);
  const event = {
    preventDefault: () => {}
  };
  wrapper.find("#user-data-form").simulate("submit", event);
  expect(propsData.onSubmit).toHaveBeenCalledWith(event);
});
