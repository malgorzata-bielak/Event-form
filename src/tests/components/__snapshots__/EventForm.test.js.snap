// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EventForm correctly 1`] = `
<Styled(form)
  autoComplete="off"
  id="user-data-form"
  onSubmit={[MockFunction]}
>
  <Input
    autofocus={true}
    event={[MockFunction]}
    id="firstname"
    labelName="First name"
  />
  <Input
    event={[MockFunction]}
    id="lastname"
    labelName="Last name"
  />
  <Input
    event={[MockFunction]}
    id="email"
    labelName="Email"
  />
  <Styled(label)
    id="date-picker"
  >
    Date
    <Styled(DatePicker)
      allowSameDay={false}
      dateFormat="MM/dd/yyyy"
      dateFormatCalendar="LLLL yyyy"
      disabled={false}
      disabledKeyboardNavigation={false}
      dropdownMode="scroll"
      inlineFocusSelectedMonth={false}
      monthsShown={1}
      nextMonthButtonLabel="Next month"
      onBlur={[Function]}
      onChange={[MockFunction]}
      onClickOutside={[Function]}
      onFocus={[Function]}
      onInputClick={[Function]}
      onInputError={[Function]}
      onKeyDown={[Function]}
      onMonthChange={[Function]}
      onSelect={[Function]}
      onYearChange={[Function]}
      preventOpenOnFocus={false}
      previousMonthButtonLabel="Previous Month"
      readOnly={false}
      renderDayContents={[Function]}
      selected={1558615381720}
      shouldCloseOnSelect={true}
      showMonthYearPicker={false}
      showTimeInput={false}
      showTimeSelect={false}
      strictParsing={false}
      timeCaption="Time"
      timeInputLabel="Time"
      timeIntervals={30}
      withPortal={false}
    />
  </Styled(label)>
  <SubmitButton />
</Styled(form)>
`;
